### ISSUE TITLE
`OrderedQueue` / `OnlineQueue` resume can fail on scalar-label buffers when shape is inferred lazily

### ISSUE THREAD

**Issue by huguesva** [rbalestr-lab/stable-pretraining#400]
`stable-pretraining` resume is not robust for `OnlineQueue` / `OrderedQueue` state when the queue is created with `shape=None` and later materialized from 1D data.

**Observed behavior**

A queue created with unknown shape starts with a placeholder buffer of shape `(max_length, 1)`. After the first append with scalar labels of shape `(B,)`, the queue materializes to `(max_length,)`.

On checkpoint resume:

- the fresh queue is recreated in placeholder shape `(max_length, 1)`
- checkpoint restore happens before the first append
- the checkpoint contains `out` with shape `(max_length,)`
- strict `load_state_dict` fails with a size mismatch

Example error:

```text
size mismatch for callbacks_modules.ordered_queue_label.out:
copying a param with shape torch.Size([20000]) from checkpoint,
the shape in current model is torch.Size([20000, 1]).

**Comment by RandallBalestriero** [comment]
https://github.com/galilai-group/stable-pretraining/commit/f770b1da22694cf8b150a3e22cb0002f4b34c6d1
